six/modules/c++/six/source/Utilities.cpp

```diff
@@ -1198,3 +1199,20 @@ void six::getErrors(const ErrorStatistics* errorStats,
         }
     }
 }
+
+std::auto_ptr<Data> six::readXML(const std::string& xmlPath, DataType dataType)
+{
+    io::FileInputStream xmlFileStream(xmlPath);
+    xml::lite::MinidomParser treeBuilder;
+    treeBuilder.parse(xmlFileStream);
+    xmlFileStream.close();
+
+    std::auto_ptr<logging::Logger> log(new logging::NullLogger());
+    six::XMLControl *control = six::XMLControlFactory::getInstance().
+            newXMLControl(dataType, log.get());
+
+    std::auto_ptr<Data> data(control->fromXML(treeBuilder.getDocument(),
+        std::vector<std::string>()));
```

Comment:
Would be good to allow the caller to optionally pass in schemas (we want them too).  How about an optional parameter that defaults to an empty vector?